openedx/adg/lms/applications/models.py

```diff
@@ -341,30 +346,76 @@ def get_user_enrolled_course(self, user):
         Returns:
             CourseOverview: Enrolled multilingual course or None
         """
-        enrolled_multilingual_course = self.multilingual_courses(
-            manager='open_multilingual_courses'
-        ).filter(
+        if user.is_anonymous:
+            return None
+
+        enrolled_multilingual_course = self.open_multilingual_courses.filter(
             course__courseenrollment__user=user,
             course__courseenrollment__is_active=True
         ).first()
         return enrolled_multilingual_course.course if enrolled_multilingual_course else None
-    # pylint: enable=no-member
 
-    def does_course_exist(self, multilingual_course):
-        return self.multilingual_courses.filter(course=multilingual_course.course).exists()
+    def get_preferred_lang_course(self):
+        """
+        Return course with preferred language.
 
-    def __str__(self):
-        return self.name
+        If course with preferred lang is not found then return the first course from the group.
+        If there are no open courses then return None.
+
+        Returns:
+            MultilingualCourse: User preferred lang course.
+        """
+        user_preferred_lang = get_language()
+        open_multilingual_courses = self.open_multilingual_courses
+
+        for multilingual_course in open_multilingual_courses:
+            if multilingual_course.course.language == user_preferred_lang:
+                return multilingual_course.course
+        return None
+
+    def get_first_course(self):
+        return self.open_multilingual_courses.first().course if self.open_multilingual_courses.exists() else None
+
+    @classmethod
+    def get_courses(cls, user, is_prereq=False):
```

Comment:
Is there any specific reason to use `cls` here? If not then can we please use `self` to make it consistent with other class methods?

Reply:
It is a ```classmethod``` that is why I have used ```cls```.